exercise_1/job_scraper.py

```diff
@@ -0,0 +1,75 @@
+import requests
+import collections
+
+
+JobOffer = collections.namedtuple(
+    "JobOffer", ("position", "company", "url", "salary", "source"),
+)
+
+
+class BaseScraper:
+    """Basic abstraction for services to handle serializing data from service
+      and wrap them around data class.
+      """
+    url = None
+    data_class = None
+
+    def get_offers(self):
+        """Get a container of API data from self.url."""
+        raise NotImplementedError
+
+    def parse_offer(self, offer):
+        """Parse single API data structure into self.data_class."""
+        raise NotImplementedError
+
+
+class NoFluffJobsScraper(BaseScraper):
+    url = "https://nofluffjobs.com/api/search/posting"
+    data_class = JobOffer
+
+    def get_offers(self):
+        return requests.get(self.url).json()["postings"]
+
+    def parse_offer(self, offer):
+        data = {
+            "position": offer.get("title"),
+            "company": offer.get("name"),
+            "url": "{}{}".format("https://nofluffjobs.com/job/", offer.get("url")),
+            "source": "nofluffjobs",
+            "salary": "-",
+        }
+        return self.data_class(**data)
+
+
+class JustJoinItScraper(BaseScraper):
+    url = "https://justjoin.it/api/offers"
+    data_class = JobOffer
+
+    def get_offers(self):
+        return requests.get(self.url).json()
+
+    def parse_offer(self, offer):
+        data = {"position": offer.get("title"),
```

Comment:
Ostatnia sprawa bo jest pięknie:
```
data = {
    "position": offer.get("title"),
     "company": offer.get("company_name"),
     "salary": "{} - {}".format(offer.get("salary_from"), offer.get("salary_to")),
     "url": "{}{}".format("https://justjoin.it/offers/", offer.get("id")),
     "source": "justjoinit",
}
```
Indentationy 4 spacje , plus 79 znaków w linii, jak definiujesz coś listę to otwierasz nawias w jednej linii i zwykle przechodzisz do kolejnej i dopiero wtedy definiujesz wpis po wpisie i zamykasz w kolejnej linii (tak jak wyżej)